### PR TITLE
Add host function with Shared Library

### PR DESCRIPTION
**Pull request by chinzhiweiblank**
#444 Create a host function for WasmEdge

This example creates a host function using the json-c shared library.

I have handwritten a parse-json.wat and used wat2wasm to create parse-json.wasm. I then use WasmEdge_VMRunWasmFromFile to execute from the file.

For the following JSON file,
```
{
    "testValue": "Success" 
}
```
it extracts the value corresponding to the key and returns it.

`Get the result: "Success"`

### CODE REVIEW

**Comment by alabulei1** [comment]
@hydai Could you help review this PR?

**Comment by hydai** [comment]
Hi @chinzhiweiblank 
I am not sure if we would like to put the examples into the `tools` folder. Maybe we should have an external path to archive, say `examples` or `docs/examples`.

To accept this PR, there are something need to fix:
1. There is no documentation or README to describe how to compile and run your application.
2. The coding style should follow `.clang-format` and `.clang-tidy`.
3. You include an external header `<json-c/json_tokener.h>`. However, there is no way to know where is this header from.

**Comment by chinzhiweiblank** [comment]
> Hi @chinzhiweiblank I am not sure if we would like to put the examples into the `tools` folder. Maybe we should have an external path to archive, say `examples` or `docs/examples`.

I have added a folder `examples/host_functions`.

> 1. There is no documentation or README to describe how to compile and run your application.
> 3. You include an external header `<json-c/json_tokener.h>`. However, there is no way to know where is this header from.

Added README.md to install the shared library `libjson`, build and run the executable.

File: examples/host_functions/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Examples
+
+The `parse-json` example shows how to use a host function with a shared library `json-c`. The `parseJson` function accepts a string as a key, parses the JSON file `test.json` and returns the corresponding value.
+
+The `parse-json.wat` is a WebAssembly script. It is compiled into WebAssembly in the form of `parse-json.wasm` using the [WABT tool](https://github.com/WebAssembly/wabt).
+
+## Installation
```
Comment (hydai) [inline examples/host_functions/README.md:7]: Could you please also mention your environment? The `libjson-c-dev` on different OS may have different version. It's better to have the detailed version information.

**Review by hydai — CHANGES_REQUESTED** [review]
I suggest that using `examples/capi/host_functions/parse_json` as your folder name.

File: examples/host_functions/README.md (hunk 1 above)
Comment (chinzhiweiblank) [reply examples/host_functions/README.md]: Added: The version of `json-c` installed is `0.13.1` in `Ubuntu 20.04.3`. Is that correct?

**Review by hydai — APPROVED** [review]
ＬＧＴＭ

**Pull request merged**